src/main/java/frc/robot/subsystems/IntakeSubsystem.java

```diff
@@ -1,19 +1,168 @@
 package frc.robot.subsystems;
 
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.DoubleSolenoid.Value;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+
+import static frc.robot.Constants.Intake.*;
 
+/**
+ * Represents the robot's dual intakes.
+ * 
+ * <p>
+ * Each intake has a double solenoid connected to two pistons and Talon 
+ * SRX connected to a BAG motor.
+ */
 public class IntakeSubsystem extends SubsystemBase {
-    public void IntakeSubsystem() {
 
+    private DoubleSolenoid leftIntakeSolenoid;
+    private DoubleSolenoid rightIntakeSolenoid;
+
+    private WPI_TalonSRX leftIntakeMotor;
+    private WPI_TalonSRX rightIntakeMotor;
+    
+    public IntakeSubsystem() {
+        leftIntakeSolenoid = new DoubleSolenoid(
+            Constants.PCM_CAN_ID,
+            Constants.PCM_TYPE,
+            LEFT_INTAKE_EXTENSION_SOLENOID_ID,
+            LEFT_INTAKE_RETRACTION_SOLENOID_ID
+        );
+
+        rightIntakeSolenoid = new DoubleSolenoid(
+            Constants.PCM_CAN_ID,
+            Constants.PCM_TYPE,
+            RIGHT_INTAKE_EXTENSION_SOLENOID_ID,
+            RIGHT_INTAKE_RETRACTION_SOLENOID_ID
+        );
+
+        leftIntakeMotor = new WPI_TalonSRX(LEFT_INTAKE_MOTOR_ID);
+        rightIntakeMotor = new WPI_TalonSRX(RIGHT_INTAKE_MOTOR_ID);
     }
 
-    @Override
-    public void periodic() {
-        // This method will be called once per scheduler run
+    /**
+     * Extend the left intake without turning on the rollers
+     */
+    public void extendLeftIntake () {
+        leftIntakeSolenoid.set(Value.kForward);
+    }
+
+    /**
+     * Extend and activate the left intake
+     */
+    public void deployLeftIntake () {
+        extendLeftIntake();
+        leftIntakeMotor.set(ControlMode.Velocity, INTAKE_MOTOR_POWER);
+    }
+
+    /**
+     * Retract the left intake and turn off its rollers
+     */
+    public void retractLeftIntake () {
+        leftIntakeSolenoid.set(Value.kReverse);
+        leftIntakeMotor.stopMotor();
+    }
+
+    /**
+     * Deploy or retract the left intake
+     * 
+     * @param deployed If true, intake will deploy, otherwise, it'll retract
+     */
+    public void setLeftIntake (boolean deployed) {
+        if (deployed) {
+            deployLeftIntake();
+        } else {
+            retractLeftIntake();
+        }
+    }
+
+    /**
+     * Extend the right intake without turning on the rollers
+     */
+    public void extendRightIntake () {
```

Comment:
Duplicated functions with the same code inside them with only a single variable change is asking for bugs. At some point you're going to change one and forget to change the other, or perhaps typo one of them.

You should combine these to a single function that takes an enum parameter to define which side you want to affect. In fact, you can roll in the both as well

```
enum Side
{
  Left,
  Right,
  Both,
}
```

Combining this with the above comment, you could simplify the API down to
```
Deploy(Side side, bool activateMotor)
{

}
```

and have the necessary internal functions to support that.